src/VisualStudio/Core/Impl/Options/AbstractOptionPage.cs

```diff
@@ -1,23 +1,39 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
 {
     [System.ComponentModel.DesignerCategory("code")] // this must be fully qualified
     internal abstract class AbstractOptionPage : UIElementDialogPage
     {
-        protected abstract AbstractOptionPageControl CreateOptionPage(IServiceProvider serviceProvider);
+        private static IOptionService s_optionService;
+        private static OptionStore s_optionStore;
+        private static bool s_needsLoadOnNextActivate = true;
+
+        protected abstract AbstractOptionPageControl CreateOptionPage(IServiceProvider serviceProvider, OptionStore optionStore);
 
         protected AbstractOptionPageControl pageControl;
```

Comment:
Not related to your change but this makes it even more confusing: can you rename this to `_pageControl` so we don't confuse it with a local?